src/text-buffer.js

```diff
@@ -1927,16 +1927,29 @@ class TextBuffer {
       try {
         await this.buffer.save(destination, this.getEncoding())
       } catch (error) {
-        const canEscalate = process.platform === 'darwin' || process.platform === 'linux'
-        if (error.code === 'EACCES' && destination === filePath && canEscalate) {
+        if (error.code !== 'EACCES' || destination !== filePath) throw error
+
+        const isMacOrLinux = process.platform === 'darwin' || process.platform === 'linux'
+        const isWindows = process.platform === 'win32'
+        if (isMacOrLinux) {
           const fsAdmin = require('fs-admin')
           try {
             await this.buffer.save(fsAdmin.createWriteStream(filePath), this.getEncoding())
           } catch (_) {
             throw error
           }
-        } else {
-          throw error
+        } else if (isWindows) {
+          const winattr = getPromisifiedWinattr()
+          const attrs = await winattr.get(filePath)
+          if (!attrs.hidden) throw error
+
+          try {
+            await winattr.set(filePath, { hidden: false })
+            await this.buffer.save(filePath, this.getEncoding())
+            await winattr.set(filePath, { hidden: true })
+          } catch (_) {
```

Comment:
```suggestion
          } catch (error) {
```

Reply:
I don't think we want to do this, as we want to throw the original error if something goes wrong